### DISCUSSION TITLE
feature: have the possiblity to hide a input parameter

### DISCUSSION THREAD

**Discussion by Jiehong** [Netflix/conductor#2416] [Ideas]
Hi,

We've seen some people use workflows with sensitive information in the parameters of the workflow.

For example, requesting a service token to call a special http service.

But anyone seeing that workflow execution can then see that token value, and that's not great.

I'm wondering if having workflow input parameters hidden could help.

For example, you could define a workflow input parameter as such:

```json
"parameter-to-hide": {
    "valueType": "string",
    "hidden": true
}
```

Now, anywhere that value is used in the workflow (with `${workflow.input.parameter-to-hide}`) would be replaced by "****" when retrieving the workflow, but the actual value would still be visible to tasks themselves.

I do see some cons:
- what if a task returns that value in its output? Should it also be hidden?
- debugging becomes harder, because now some values are hidden

What do you think?

If you have a better idea for such a usecase, feel free to explain it!

**Comment by kishorebanala** [comment]
@Jiehong Given the bespoke nature of this feature, and the number of gotcha's we need to consider in this implementation, we think it's best not to implement this in Conductor. However, if this is something you only need in the UI, that might be rather simple to implement.

**Comment by Jiehong** [comment]
@kishorebanala : fair enough, I agree with the big extra complexity required for this feature.
